crates/bevy_render/src/texture/hdr_texture_loader.rs

```diff
@@ -23,6 +24,10 @@ pub enum HdrTextureLoaderError {
     Io(#[from] std::io::Error),
     #[error("Could not extract image: {0}")]
     Image(#[from] image::ImageError),
+    #[error("Could not convert total number of bytes in decoded image from u64 to usize: {0}")]
+    TryIntoTotalBytes(#[from] std::num::TryFromIntError),
```

Comment:
I would wait for other opinions before acting on this, but IMO this could be removed and replaced with `as usize` below.

This is a conversion from u64 to usize, so the failure mode is users on 32 bit platforms attempting to decode a 4 gig hdr file.

Reply:
Don't worry, your suggestion to allocate the capacity worked perfectly so I no longer need to add this error.